cli/src/pcluster/api/awslambda/entrypoint.py

```diff
@@ -52,8 +51,6 @@ def lambda_handler(event: Dict[str, Any], context: LambdaContext) -> Dict[str, A
             # Instrument X-Ray recorder to trace requests served by the Flask application
             xray_recorder.configure(service="ParallelCluster Flask App")
             XRayMiddleware(pcluster_api.flask_app, xray_recorder)
-        # Clearing cache since the same ParallelClusterFlaskApp is reused
-        Cache.clear_all()
```

Comment:
Correct! That was the initial use case addressed by the clear call in the Lambda entrypoint. I then moved this logic to the Flask API so that any consumer that re-uses the application across multiple requests gets a correct behavior.